frontend/src/modules/Maintenance/components/MaintenanceViewer.js

```diff
@@ -314,11 +309,32 @@ export const MaintenanceViewer = () => {
     useState(START_MAINTENANCE);
   const [maintenanceWindowStatus, setMaintenanceWindowStatus] =
     useState(INACTIVE_STATUS);
+  const [maintenanceModes, setMaintenanceModes] = useState([]);
   const [dropDownStatus, setDropDownStatus] = useState(false);
   const [refreshingTimer, setRefreshingTimer] = useState('');
   const { enqueueSnackbar, closeSnackbar } = useSnackbar();
   const dispatch = useDispatch();
 
+  const fetchMaintenanceModes = useCallback(async () => {
```

Comment:
I am no FE expert but I am bit concerned that this might increase the latency and make the UI laggy (imagine waiting 100-200ms for a dropdown to populate). I think we might need caching or prepopulating the fields during packaging. I am happy to not do it now but I think we should generalise the enum fetching to a util `fetchEnum('MaintenanceModes', setMaintenanceModes)` that will allow us to customise the way we fetch later without changing the client code. wdyt?

Reply:
I like the idea of optimisation. I will try to do it, but I'm not sure I am good enough in react)